client/src/main/java/io/spine/client/FilterMixin.java

```diff
@@ -0,0 +1,147 @@
+        FieldPath fieldPath = getFieldPath();
+        return Field.withPath(fieldPath);
+    }
+
+    /**
+     * Checks if the target field is present in the specified message type.
+     */
+    default boolean fieldPresentIn(Descriptor message) {
+        checkNotNull(message);
+        Field field = field();
+        boolean result = field.presentIn(message);
+        return result;
+    }
+
+    /**
+     * Verifies that the target field is present in the passed message type.
+     */
+    default void checkFieldPresentIn(Descriptor message) {
+        checkNotNull(message);
+        if (!fieldPresentIn(message)) {
+            throw newIllegalArgumentException(
+                    "The field with path `%s` is not present in the message type `%s`.",
+                    field(), message.getFullName());
+        }
+    }
+
+    /**
+     * Checks if the target field is an entity column in the passed message type.
+     */
+    default boolean fieldIsColumnIn(Descriptor message) {
+        checkNotNull(message);
+        Optional<FieldDescriptor> fieldDescriptor = field().findDescriptor(message);
+        if (!fieldDescriptor.isPresent()) {
+            return false;
+        }
+        FieldDeclaration declaration = new FieldDeclaration(fieldDescriptor.get());
+        boolean result = isColumn(declaration);
+        return result;
+    }
+
+    /**
+     * Verifies that the target field is an entity column in the passed message type.
+     */
+    default void checkFieldIsColumnIn(Descriptor message) {
+        checkNotNull(message);
+        checkFieldIsTopLevel();
+        if (!fieldIsColumnIn(message)) {
+            throw newIllegalArgumentException(
+                    "The entity column `%s` is not found in entity state type `%s`. " +
+                            "Please check the field exists and is marked with `(column)` option.",
+                    field(), message.getFullName());
+        }
+    }
+
+    /**
+     * Checks if the target field is a top-level field.
+     */
+    default boolean fieldIsTopLevel() {
+        return !field().isNested();
+    }
+
+    /**
+     * Verifies that the target field is a top-level field.
+     */
+    default void checkFieldIsTopLevel() {
+        if (!fieldIsTopLevel()) {
+            throw newIllegalArgumentException(
+                    "The entity filter contains a nested entity column `%s`. " +
+                            "Nested entity columns are currently not supported.",
+                    field()
+            );
+        }
+    }
+
+    /**
+     * Validates a filter against the queried type.
+     *
+     * <p>Makes sure the target field is a valid entity column in case the {@link EntityState} is
+     * queried and is a valid message field in all other cases.
+     *
+     * @throws IllegalArgumentException
+     *         if the target field is not present in the type or doesn't satisfy the constraints
+     */
+    default void validateAgainst(TypeUrl targetType) {
```

Comment:
This is way too broad. It looks closer to `checkCanFilter(TypeUrl targetType)`. It would be even better if `checkCanApplyTo(Target target)`, but I am not sure that's possible.